File: Easy/2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.java
class Solution{
    public int minimumCost(int[] cost){
        int n=cost.length;;

        Integer[] c=Arrays.stream(cost)
                            .boxed()
                            .sorted(Collections.reverseOrder())
                            .toArray(Integer[]::new);
        int sum=0;
        int i=0;
        while(i<n){
            sum += c[i];
            if(i+1<n){
                sum += c[i+1];
            }

            i+=3;
        }

        return sum;
    }
}

File: Easy/2248-minimum-cost-of-buying-candies-with-discount/minimum-cost-of-buying-candies-with-discount.py
class Solution:
    def minimumCost(self, c: List[int]) -> int:
        n=len(c)

        c.sort(reverse=True)
        sum=0
        i=0
        while i<n:
            sum += c[i]
            if i+1<n:
                sum += c[i+1]

            i+=3
    
        return sum
